framework/src/base/MooseApp.C

```diff
@@ -88,240 +89,250 @@ void
+  params.addCommandLineParam<MooseEnum>(
+      "color", "--color <auto,on,off=on>", colors, "Whether to use color in console output");
+  params.setGlobalCommandLineParam("color");
+
+  params.addCommandLineParam<bool>("help", "-h --help", "Displays CLI usage statement");
   params.addCommandLineParam<bool>(
       "minimal",
       "--minimal",
-      false,
-      "Ignore input file and build a minimal application with Transient executioner.");
+      "Ignore input file and build a minimal application with Transient executioner");
 
   params.addCommandLineParam<bool>(
       "language_server",
       "--language-server",
       "Starts a process to communicate with development tools using the language server protocol");
 
-  params.addCommandLineParam<std::string>(
+  params.addCommandLineParam<bool>(
       "definition", "--definition", "Shows a SON style input definition dump for input validation");
+  params.addCommandLineParam<bool>("dump", "--dump", "Shows a dump of available input file syntax");
   params.addCommandLineParam<std::string>(
-      "dump", "--dump [search_string]", "Shows a dump of available input file syntax.");
-  params.addCommandLineParam<bool>(
-      "registry", "--registry", "Lists all known objects and actions.");
+      "dump_search",
+      "--dump-search <search>",
+      "Shows a dump of available input syntax matching a search");
+  params.addCommandLineParam<bool>("registry", "--registry", "Lists all known objects and actions");
   params.addCommandLineParam<bool>(
-      "registry_hit", "--registry-hit", "Lists all known objects and actions in hit format.");
+      "registry_hit", "--registry-hit", "Lists all known objects and actions in hit format");
   params.addCommandLineParam<bool>(
-      "use_executor", "--executor", false, "Use the new Executor system instead of Executioners");
+      "use_executor", "--executor", "Use the new Executor system instead of Executioners");
 
   params.addCommandLineParam<bool>(
-      "apptype", "--type", false, "Return the name of the application object.");
+      "show_type", "--show-type", "Return the name of the application object");
+  params.addCommandLineParam<bool>("yaml", "--yaml", "Dumps all input file syntax in YAML format");
   params.addCommandLineParam<std::string>(
-      "yaml", "--yaml", "Dumps input file syntax in YAML format.");
+      "yaml_search", "--yaml-search", "Dumps input file syntax matching a search in YAML format");
+  params.addCommandLineParam<bool>("json", "--json", "Dumps all input file syntax in JSON format");
   params.addCommandLineParam<std::string>(
-      "json", "--json", "Dumps input file syntax in JSON format.");
+      "json_search", "--json-search", "Dumps input file syntax matching a search in JSON format");
   params.addCommandLineParam<bool>(
-      "syntax", "--syntax", false, "Dumps the associated Action syntax paths ONLY");
+      "syntax", "--syntax", "Dumps the associated Action syntax paths ONLY");
```

Comment:
No more default to `false`... Is that because the default is the parameter not being specified equalling to false?

Reply:
Because a bool CLI param can only ever be false (and being set making it true), setting the value of false or setting no value is equivalent